CRM/Activity/BAO/Activity.php

```diff
@@ -656,246 +656,199 @@ public static function logActivityAction($activity, $logMessage = NULL) {
-    $query = "
-INSERT INTO {$activityContactTempTable} ( activity_id, contact_id, record_type_id, contact_name, is_deleted )
-SELECT     ac.activity_id,
-           ac.contact_id,
-           ac.record_type_id,
-           c.sort_name,
-           c.is_deleted
-FROM       {$activityTempTable}
-INNER JOIN civicrm_activity a ON ( a.id = {$activityTempTable}.activity_id )
-INNER JOIN civicrm_activity_contact ac ON ( ac.activity_id = {$activityTempTable}.activity_id )
-INNER JOIN civicrm_contact c ON c.id = ac.contact_id
-WHERE ac.record_type_id != %1
-";
-    $params = array(1 => array($targetID, 'Integer'));
-    CRM_Core_DAO::executeQuery($query, $params);
-
-    $activityFields = array("ac.activity_id", "ac.contact_id", "ac.record_type_id", "c.sort_name", "c.is_deleted");
-    $select = CRM_Contact_BAO_Query::appendAnyValueToSelect($activityFields, "ac.activity_id");
-
-    // for each activity insert one target contact
-    // if we load all target contacts the performance will suffer a lot for mass-activities.
-    $query = "
-INSERT INTO {$activityContactTempTable} ( activity_id, contact_id, record_type_id, contact_name, is_deleted, counter )
-{$select}, count(ac.contact_id)
-FROM       {$activityTempTable}
-INNER JOIN civicrm_activity a ON ( a.id = {$activityTempTable}.activity_id )
-INNER JOIN civicrm_activity_contact ac ON ( ac.activity_id = {$activityTempTable}.activity_id )
-INNER JOIN civicrm_contact c ON c.id = ac.contact_id
-WHERE ac.record_type_id = %1
-GROUP BY ac.activity_id
-";
-
-    CRM_Core_DAO::executeQuery($query, $params);
+    if (empty($order)) {
+      // context = 'activity' in Activities tab.
+      $activityParams['options']['sort'] = (CRM_Utils_Array::value('context', $params) == 'activity') ? "activity_date_time DESC" : "status_id ASC, activity_date_time ASC";
+    }
+    else {
+      $activityParams['options']['sort'] = str_replace('activity_type ', 'activity_type_id.label ', $order);
+    }
 
-    // step 3: Combine all temp tables to get final query for activity selector
-    // sort by the original sort order, stored in fixed_sort_order
-    $query = "
-SELECT     {$activityTempTable}.*,
-           {$activityContactTempTable}.contact_id,
-           {$activityContactTempTable}.record_type_id,
-           {$activityContactTempTable}.contact_name,
-           {$activityContactTempTable}.is_deleted,
-           {$activityContactTempTable}.counter,
-           re.parent_id as is_recurring_activity
-FROM       {$activityTempTable}
-INNER JOIN {$activityContactTempTable} on {$activityTempTable}.activity_id = {$activityContactTempTable}.activity_id
-LEFT JOIN civicrm_recurring_entity re on {$activityContactTempTable}.activity_id = re.entity_id
-ORDER BY    fixed_sort_order
-        ";
+    //TODO :
+    // 1. missing support to sort by activity_type in API,
+    //    this will cause regression in selector column if someone sort by activity type
+    $result = civicrm_api3('Activity', 'Get', $activityParams);
```

Comment:
Hmm - I think you should add some comments about that - it seems like the api could do that filtering - it just isn't right now, let's put some @todos for the future